fe/fe-core/src/main/java/com/starrocks/catalog/Column.java

```diff
@@ -229,6 +234,55 @@ public void setIsAllowNull(boolean isAllowNull) {
         this.isAllowNull = isAllowNull;
     }
 
+    public String getMetaDefaultValue(List<String> extras) {
+        String defaultValue = FeConstants.null_string;
+        if (this.getDefaultValue() != null) {
+            defaultValue = this.getDefaultExpr();
+        } else if (this.getDefaultExpr() != null) {
+            if ("now()".equalsIgnoreCase(this.getDefaultExpr())) {
+                defaultValue = "CURRENT_TIMESTAMP";
+                extras.add("DEFAULT_GENERATED");
+            }
+        }
+        return defaultValue;
+    }
+
+    public String getDefaultExpr() {
+        return defaultExpr;
+    }
+
+    public void setDefaultExpr(String defaultExpr) {
+        this.defaultExpr = defaultExpr;
+    }
+
+    // if the value or expr calculated is the same for each row of a batch like now(). return true
+    // else for a batch of every row different like uuid(). return null
+    public boolean hasDefaultValue() {
+        if (defaultValue != null) {
+            return true;
+        }
+        if ("now()".equalsIgnoreCase(defaultExpr)) {
+            return true;
+        }
+        return false;
+    }
+
+    // if the value or expr calculated is the same for each row of a batch like now(). calculated default value
+    // else for a batch of every row different like uuid(). return null
+    public String getCalculatedDefaultValue() {
+        if (defaultValue != null) {
+            return defaultValue;
+        }
+
+        if (!isEvaluate) {
+            if ("now()".equalsIgnoreCase(defaultExpr)) {
+                evaluatedDefaultValue = DateTime.now().toString();
+            }
+            isEvaluate = true;
+        }
+        return evaluatedDefaultValue;
```

Comment:
I'm sorry that the calculated value can not be cached.
Because default value is different for different users.
For example, there are two load jobs that need to get default value for this column. They started at different time, they should get different default values for each. If you cache the calculated value, they will get the same default value, which is not expected.